sqld/src/hrana/proto.rs

```diff
@@ -84,6 +86,17 @@ pub struct BatchResp {
     pub result: BatchResult,
 }
 
+#[derive(Deserialize, Debug)]
+pub struct SequenceReq {
+    pub stream_id: i32,
+    #[serde(default)]
+    pub sql: Option<String>,
+    #[serde(default)]
+    pub sql_id: Option<i32>,
```

Comment:
Is there a way to make serde deserialize sql and sql_id fields into an Enum?

Reply:
It might be possible to do this by using [`#[serde(flatten)]`](https://serde.rs/field-attrs.html#flatten) and some manual serializer/deserializer implementation, but I decided to keep the structures simple. There is a helper method that extracts the SQL text from `sql` xor `sql_id` or produces an error if not exactly one field was given.